libethcore/TransactionBase.cpp

```diff
@@ -64,23 +64,25 @@ TransactionBase::TransactionBase(bytesConstRef _rlpData, CheckTransaction _check
 
 		m_data = rlp[5].toBytes();
 
-		byte v = rlp[6].toInt<byte>();
-		h256 r = rlp[7].toInt<u256>();
-		h256 s = rlp[8].toInt<u256>();
+		int const v = rlp[6].toInt<int>();
+		h256 const r = rlp[7].toInt<u256>();
+		h256 const s = rlp[8].toInt<u256>();
 
-		m_vrs = SignatureStruct{ r, s, v };
-
-		if (hasZeroSignature())
-			m_chainId = m_vrs->v;
+		if (isZeroSignature(r, s))
+		{
+			m_chainId = static_cast<int>(v);
+			m_vrs = SignatureStruct{r, s, 0};
+		}
 		else
 		{
 			if (v > 36)
-				m_chainId = (v - 35) / 2;
+				m_chainId = static_cast<int>((v - 35) / 2); 
```

Comment:
Why do you need a cast here?  `v` is `int`, `35` and `2` are both `int`, I think.

Reply:
because of the division?

Reply:
@pirapira You are right, it's not needed, I'll remove it